### PR TITLE
APIS-4929 - Added UserId to Collaborator

### PR DESCRIPTION
**Pull request by anjumabbas5**

### CODE REVIEW

File: app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.UUID
+
+case class UserId(value: String) extends AnyVal
+
+object UserId {
+  import play.api.libs.json.Json
+  implicit val userIdFormat = Json.valueFormat[UserId]
+
+  def random: Option[UserId] = Some(UserId(UUID.randomUUID().toString))
+}
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala:28]: newline

File: app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.thirdpartyapplication.models
+
+import java.util.UUID
+
+case class UserId(value: String) extends AnyVal
```
Comment (peteslater-ee) [inline app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala:21]: Should UserId be a string or a UUID?

Comment (AndySpaven) [reply app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala:21]: We can change this afterwards when we're also ready to apply the APP_ID PR

Comment (anjumabbas5) [reply app/uk/gov/hmrc/thirdpartyapplication/models/UserId.scala:21]: UUID probably. However it is consistent with how ApplicationId currently works.

**Review by peteslater-ee — APPROVED** [review]

**Pull request merged**